backend/services/syson-model-services/src/main/java/org/eclipse/syson/model/services/ModelMutationElementService.java

```diff
@@ -69,47 +72,92 @@ public Element setAsView(ViewUsage viewUsage, String newViewDefinition) {
-            EObject duplicatedObject = copier.copyWithoutContent(elementToDuplicate);
-            if (copyOutgoingReferences) {
-                copier.copyReferences();
-            }
-            duplicateObject = Optional.ofNullable(duplicatedObject);
+    public Element createPartUsageAndBindingConnectorAsUsage(PartUsage self) {
+        var parent = self.getOwner();
+        if (parent instanceof Namespace parentNamespace) {
+            // create a new port on given part usage
+            var newSelfPort = SysmlFactory.eINSTANCE.createPortUsage();
+            this.metamodelMutationElementService.addChildInParent(self, newSelfPort);
+            this.metamodelMutationElementService.initialize(newSelfPort);
+            // create a new part usage as a self sibling
+            var newPartUsage = SysmlFactory.eINSTANCE.createPartUsage();
+            this.metamodelMutationElementService.addChildInParent(parent, newPartUsage);
+            this.metamodelMutationElementService.initialize(newPartUsage);
+            // create a new port on the new part usage
+            var newPartUsagePort = SysmlFactory.eINSTANCE.createPortUsage();
+            this.metamodelMutationElementService.addChildInParent(newPartUsage, newPartUsagePort);
+            this.metamodelMutationElementService.initialize(newPartUsagePort);
+            // create binding connector as usage edge between both new ports
+            this.metamodelMutationElementService.createBindingConnectorAsUsage(newSelfPort, newPartUsagePort, parentNamespace);
+            return newPartUsage;
         }
+        return self;
+    }
 
-        return duplicateObject.filter(Element.class::isInstance)
-                .map(Element.class::cast)
-                .map(duplicate -> {
-                    // Change the name of element root element (not its content)
-                    if (elementToDuplicate.getDeclaredName() != null && !elementToDuplicate.getDeclaredName().isBlank()) {
-                        duplicate.setDeclaredName(elementToDuplicate.getDeclaredName() + "-copy");
-                    } else if (elementToDuplicate.getShortName() != null && !elementToDuplicate.getShortName().isBlank()) {
-                        duplicate.setDeclaredShortName(elementToDuplicate.getDeclaredShortName() + "-copy");
-                    }
-                    // Reset all ids
-                    EMFUtils.allContainedObjectOfType(duplicate, Element.class).forEach(element -> element.setElementId(ElementUtil.generateUUID(element).toString()));
+    /**
+     * Create a sibling {@link PartUsage} and then connect self and the newly created {@link PartUsage} through an {@link org.eclipse.syson.sysml.FlowUsage} connected to 2 new
+     * {@link org.eclipse.syson.sysml.PortUsage}.
+     *
+     * @param self
+     *         a {@link PartUsage}
+     * @return the newly create {@link PartUsage}
+     */
+    public PartUsage createPartUsageAndFlowConnection(PartUsage self) {
```

Comment:
Only moved from ViewCreateService and redirect some sub methods to their new location this.metamodelMutationElementService